core/src/main/java/org/testcontainers/utility/ImagePullCountLogger.java

```diff
@@ -0,0 +1,50 @@
+package org.testcontainers.utility;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+/**
+ * Simple utility to log which images have been pulled by {@link org.testcontainers.Testcontainers} and how many times.
+ */
+@Slf4j
+public class ImagePullCountLogger {
+
+    private static ImagePullCountLogger instance;
+    private final Map<String, AtomicInteger> pullCounters = new HashMap<>();
+
+    public synchronized static ImagePullCountLogger instance() {
+        if (instance == null) {
+            instance = new ImagePullCountLogger();
+            Runtime.getRuntime().addShutdownHook(new Thread(instance::logStatistics));
```

Comment:
Using a JVM shutdown hook. This is not the most iron-clad way, but as this logging is informational and not required I believe it's sufficient.